ksqldb-test-util/src/main/java/io/confluent/ksql/test/util/KafkaEmbedded.java

```diff
@@ -104,7 +106,9 @@ String brokerList() {
    * @return the broker list
    */
   String brokerList(final SecurityProtocol securityProtocol) {
-    return kafka.config().hostName() + ":"
+    final EndPoint endPoint = kafka.advertisedListeners().head();
```

Comment:
This doesn't seem correct. You want to look through the listeners and find the right one. The endpoint port will be correct, so you don't have to call `kafka.boundPort` at all.

Reply:
Hey @ijuma thanks for chiming in. To clarify your suggestion: are you saying that `kafka.advertisedListeners()` should already contain endpoints for all the different security protocols, and so this method should simply iterate through them in order to find the right one, rather than trying to construct a new endpoint? If so I've made the change in https://github.com/confluentinc/ksql/pull/7731.

The old behavior was not incorrect for the tests in this codebase since the host was the same for the different endpoints in all cases where this method was called, but I have updated the logic according to your suggestion anyway so this method is more correct going forward.

Reply:
Yes, that's right. Thanks for the follow up PR.